app/products/calls/alerts.ts

```diff
@@ -119,33 +119,38 @@ export const leaveAndJoinWithAlert = async (
             }, {leaveChannelName, joinChannelName});
         }
 
-        Alert.alert(
-            formatMessage({
-                id: 'mobile.leave_and_join_title',
-                defaultMessage: 'Are you sure you want to switch to a different call?',
-            }),
-            joinMessage,
-            [
-                {
-                    text: formatMessage({
-                        id: 'mobile.post.cancel',
-                        defaultMessage: 'Cancel',
-                    }),
-                    style: 'destructive',
-                },
-                {
-                    text: formatMessage({
-                        id: 'mobile.leave_and_join_confirmation',
-                        defaultMessage: 'Leave & Join',
-                    }),
-                    onPress: () => doJoinCall(joinServerUrl, joinChannelId, joinChannelIsDMorGM, newCall, intl, title, rootId),
-                    style: 'cancel',
-                },
-            ],
-        );
-    } else {
-        doJoinCall(joinServerUrl, joinChannelId, joinChannelIsDMorGM, newCall, intl, title, rootId);
+        const asyncAlert = async () => new Promise((resolve) => {
+            Alert.alert(
+                formatMessage({
+                    id: 'mobile.leave_and_join_title',
+                    defaultMessage: 'Are you sure you want to switch to a different call?',
+                }),
+                joinMessage,
+                [
+                    {
+                        text: formatMessage({
+                            id: 'mobile.post.cancel',
+                            defaultMessage: 'Cancel',
+                        }),
+                        onPress: async () => resolve(false),
+                        style: 'destructive',
+                    },
+                    {
+                        text: formatMessage({
+                            id: 'mobile.leave_and_join_confirmation',
+                            defaultMessage: 'Leave & Join',
+                        }),
+                        onPress: async () => resolve(await doJoinCall(joinServerUrl, joinChannelId, joinChannelIsDMorGM, newCall, intl, title, rootId)),
+                        style: 'cancel',
```

Comment:
if this is the primary action to join, why style is `cancel`? I think you want to use the `isPreferred` property instead ?